Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebExtension.cpp

```diff
@@ -172,6 +210,27 @@ static void webkit_web_extension_class_init(WebKitWebExtensionClass* klass)
         g_cclosure_marshal_generic,
         G_TYPE_NONE, 1,
         WEBKIT_TYPE_USER_MESSAGE);
+
+    /**
+     * WebKitWebExtension::initialize-web-extensions:
+     * @extension: the #WebKitWebExtension on which the signal is emitted
+     * @userData: the #GVariant object received
+     *
+     * This signal is emitted when a main resource load is initiated. The
+     * #GVariant user data comes from a webkit_web_view_set_web_extensions_initialization_user_data()
+     * called over the #WebKitWebView corresponding to @extension.
```

Comment:
Note: this doesn't make sense because all web views in this web context correspond to this extension. (This problem would go away if you added a new page-created signal instead.)

Reply:
That could be a wrong assumption. Since 2.26 [1][2], where WEBKIT_PROCESS_MODEL_SHARED_SECONDARY_PROCESS is set deprecated, the Web context can be set with WEBKIT-PROCESS-MODEL-MULTIPLE-SECONDARY-PROCESSES . I that scenario every single WebView uses a dedicated WebProcess.


[1] https://webkitgtk.org/reference/webkit2gtk/stable/WebKitWebContext.html
[2] https://webkitgtk.org/reference/webkit2gtk/stable/WebKitWebContext.html#WEBKIT-PROCESS-MODEL-MULTIPLE-SECONDARY-PROCESSES:CAPS

Reply:
I see your point. Still the signal can be useful for sending initialization data for the WebExtension from the UI-process for every single WebView separately but I am also agree with a `page-created-with-user-data` signal associated with the WebPage it could be even more useful.